src/ng/browser.js

```diff
@@ -235,10 +235,22 @@ function Browser(window, document, $log, $sniffer) {
 
   // This variable should be used *only* inside the cacheState function.
   var lastCachedState = null;
+  function getCurrentState() {
+    try {
+      var state = history.state;
+      if (state === void 0) {
```

Comment:
Since `history.state` is either going to be a valid object, undefined or null, or it is going to throw an error we can simply do:

```js
function getCurrentState() {
  try {
    return history.state || null;
  } catch(e) {
    return null;
  }
}
```